src/test/java/org/nuxeo/wopi/lock/TestLock.java

```diff
@@ -0,0 +1,194 @@
+import static org.nuxeo.wopi.Constants.LOCK_TTL;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventContext;
+import org.nuxeo.ecm.core.event.EventProducer;
+import org.nuxeo.ecm.core.event.impl.EventContextImpl;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+import org.nuxeo.wopi.FileInfo;
+
+/**
+ * Lock related tests.
+ *
+ * @since 10.3
+ */
+@RunWith(FeaturesRunner.class)
+@Features(PlatformFeature.class)
+@Deploy("org.nuxeo.ecm.platform.web.common")
+@Deploy("org.nuxeo.wopi.rest.api")
+public class TestLock {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected EventProducer eventProducer;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    protected DocumentModel doc;
+
+    protected DocumentModel expiredLockDoc;
+
+    protected String fileId;
+
+    protected String expiredLockFileId;
+
+    @Before
+    public void setUp() {
+        doc = session.createDocumentModel("/", "doc", "File");
+        doc = session.createDocument(doc);
+        fileId = FileInfo.computeFileId(doc, FILE_CONTENT_PROPERTY);
+
+        expiredLockDoc = session.createDocumentModel("/", "expiredLockDoc", "File");
+        expiredLockDoc = session.createDocument(expiredLockDoc);
+        expiredLockFileId = FileInfo.computeFileId(expiredLockDoc, FILE_CONTENT_PROPERTY);
+    }
+
+    @Test
+    public void testLockDirectory() throws InterruptedException {
+        // get unknown lock
+        assertNull(LockHelper.getLock("unknownFileId"));
+
+        // add lock / get lock
+        LockHelper.addLock(fileId, "wopiLock");
+        assertEquals("wopiLock", LockHelper.getLock(fileId));
+
+        // update lock
+        LockHelper.updateLock(fileId, "updatedWopiLock");
+        assertEquals("updatedWopiLock", LockHelper.getLock(fileId));
+
+        // refresh lock
+        LockHelper.doPriviledgedOnLockDirectory(session -> {
+            try {
+                long timestamp = (long) session.getEntry(fileId).getProperty(LOCK_DIRECTORY_SCHEMA_NAME,
+                        LOCK_DIRECTORY_TIMESTAMP);
+                Thread.sleep(10);
+                LockHelper.refreshLock(fileId);
+                long refreshedTimestamp = (long) session.getEntry(fileId).getProperty(LOCK_DIRECTORY_SCHEMA_NAME,
+                        LOCK_DIRECTORY_TIMESTAMP);
+                assertTrue(Instant.ofEpochMilli(refreshedTimestamp).isAfter(Instant.ofEpochMilli(timestamp)));
```

Comment:
Why not just `refreshedTimestamp > timestamp` ?

Reply:
Right :+1: